packages/zowe-explorer-api/src/utils/ErrorCorrelator.ts

```diff
@@ -0,0 +1,360 @@
+                    },
+                    {
+                        matches: ["File not found."],
+                        summary: "The specified UNIX file cannot be found. Perhaps the folder or file path was incorrectly specified.",
+                        tips: ["Ensure that the UNIX folder or file path is correct and try again."],
+                    },
+                ],
+            },
+        ],
+        [
+            ZoweExplorerApiType.Jes,
+            {
+                zosmf: [
+                    {
+                        matches: ["No job found for reference:"],
+                        summary: "The job modification request specified a job that does not exist.",
+                        tips: [],
+                    },
+                    {
+                        matches: ["Submit input data does not start with a slash"],
+                        summary: "The first character for the submitted job is invalid - expected a slash.",
+                        tips: ["Ensure that the input data set or file contains EBCDIC data"],
+                    },
+                    {
+                        matches: ["Job input was not recognized by system as a job"],
+                        summary: "The job was submitted without a job statement or with unrecognized (non-JCL) content.",
+                    },
+                    {
+                        errorCode: "400",
+                        matches: ["Value of jobid query parameter is not valid"],
+                        summary: "The given Job ID is invalid. Please verify that the job ID is correct and try again.",
+                    },
+                ],
+            },
+        ],
+        [
+            ZoweExplorerApiType.All,
+            {
+                any: [
+                    {
+                        errorCode: "401",
+                        matches: ["Token is not valid or expired"],
+                        summary:
+                            // eslint-disable-next-line max-len
+                            "Your connection is no longer active for profile {{profileName}}. Please log in to an authentication service to restore the connection.",
+                    },
+                    {
+                        errorCode: "401",
+                        matches: ["Username or password are not valid or expired", "All configured authentication methods failed"],
+                        summary:
+                            // eslint-disable-next-line max-len
+                            "Invalid credentials for profile {{profileName}}. Please ensure the username and password are valid or this may lead to a lock-out.",
+                    },
+                ],
+            },
+        ],
+    ]);
+
+    private constructor() {}
+
+    public static getInstance(): ErrorCorrelator {
+        if (!ErrorCorrelator.instance) {
+            ErrorCorrelator.instance = new ErrorCorrelator();
+        }
+
+        return ErrorCorrelator.instance;
```

Comment:
I know we've used `get` in front of our properties (`getApi...`, `getInstance`, ... ).
I'm curious if it makes sense to take advantage of getters (even though they can be a bit of a pain to mock during tests).

If we do decide to use getter, it may make the use a bit better for  extenders, i.e. they will be able to just do `ErrorCorrelator.instance` (similar to what Imperative does with many of its classes, like ImperativeConfig.instance and others)

Reply:
Not to play devil's advocate here because I agree that we should have getters with just the name `instance`. However for consistency's sake, I think it makes sense to keep it as `getInstance`, as we have other singletons in ZE and ZE API (table facilities) with the same pattern. We could address this across all our singletons in v4 😋

Reply:
🙏 
yeah, I `getIt()` 😋 
Make sense to keep things consistent 🙏

Reply:
Created:
- https://github.com/zowe/zowe-explorer-vscode/issues/3284